gst/tensor_transform/tensor_transform.h

```diff
@@ -66,23 +66,45 @@ typedef struct _GstTensor_Transform GstTensor_Transform;
 
 typedef struct _GstTensor_TransformClass GstTensor_TransformClass;
 
-typedef enum {
-  GTT_DIMCHG = 0, /* Dimension Change. "dimchg" */
-  GTT_TYPECAST = 1, /* Type change. "typecast" */
-
+typedef enum
+{
+  GTT_DIMCHG = 0,               /* Dimension Change. "dimchg" */
+  GTT_TYPECAST = 1,             /* Type change. "typecast" */
+  GTT_ARITHMETIC = 2,           /* Type change. "typecast" */
 
   GTT_END,
 } tensor_transform_mode;
 
+typedef enum
+{
+  ARITH_ADD = 0,
+  ARITH_MUL = 1,
+  ARITH_END,
+} tensor_transform_arith_mode;
+
+/**
+ * @brief Internal data structure for dimchg mode.
+ */
 typedef struct _tensor_transform_dimchg {
   int from;
   int to;
 } tensor_transform_dimchg;
 
+/**
+ * @brief Internal data structure for typecast mode.
+ */
 typedef struct _tensor_transform_typecast {
   tensor_type to; /**< tensor_type after cast. _NNS_END if unknown */
 } tensor_transform_typecast;
 
+/**
+ * @brief Internal data structure for arithmetic mode.
+ */
+typedef struct _tensor_transform_arithmetic {
+  tensor_transform_arith_mode mode;
+  double value;
```

Comment:
Would it safe to assume that "double" will be sufficient for all needs?
(Are we going to do floating-point operations even if the input/output is integer and the user is adding "1"?)

Reply:
I think this value is the buffer to hold various type and that's why the double is used. I will use int64 instead and put TODO comment.